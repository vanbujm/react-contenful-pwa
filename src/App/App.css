.App {
    text-align: center;
    display: flex;
    flex-direction: column;
    flex: 1;
    background-color: #c7c7c7;
}

.App-logo {
    animation: App-logo-spin infinite 20s linear;
    width: 212px;
    height: 150px;
}

.App-header {
    background-color: #282c34;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-start;
    font-size: calc(10px + 2vmin);
    color: white;
}

.App-link {
    color: #61dafb;
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}

.link {
    margin: 0 1rem;
    color: white;
    text-decoration: none;
}

.link:hover {
    color: #c7c7c7;
}
