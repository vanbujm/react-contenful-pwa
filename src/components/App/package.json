{
  "main": "App.js",
  "name": "App",
  "version": "0.0.0",
  "private": true
}