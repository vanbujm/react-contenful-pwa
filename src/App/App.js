import React from 'react';
import { BrowserRouter, Link, Route, Switch } from 'react-router-dom';

import Layout from '../Layout';
import Home from '../Home';
import Content from '../Content';
import logo from '../logo.svg';
import './App.css';

const App = () => (
  <BrowserRouter>
    <div className="App">
      <header className="App-header">
        <Link to="/">
          <img src={logo} className="App-logo" alt="logo" />
        </Link>
        <Link className="link" to="/content">
          Content
        </Link>
      </header>
      <Layout>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/content" component={Content} />
        </Switch>
      </Layout>
    </div>
  </BrowserRouter>
);

export default App;
