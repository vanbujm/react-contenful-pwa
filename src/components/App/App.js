import React from 'react';
import { BrowserRouter, Link, Route, Switch } from 'react-router-dom';
import Loadable from 'react-loadable';

import Layout from '../Layout';
import logo from '../../logo.svg';
import './App.css';
import Loading from '../Loading';

const loadingConfig = {
  delay: 100,
  timeout: 10000
};

const Content = Loadable({
  loader: () => import('../Content'),
  loading: Loading,
  ...loadingConfig
});

const Home = Loadable({
  loader: () => import('../Home'),
  loading: Loading,
  ...loadingConfig
});

const App = () => (
  <BrowserRouter>
    <div className="App">
      <header className="App-header">
        <Link to="/">
          <img src={logo} className="App-logo" alt="logo" />
        </Link>
        <Link className="link" to="/content">
          Content
        </Link>
      </header>
      <Layout>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/content" component={Content} />
        </Switch>
      </Layout>
    </div>
  </BrowserRouter>
);

export default App;
