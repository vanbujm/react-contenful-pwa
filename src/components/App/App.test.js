import React from 'react';
import App from './App';
import TestRenderer from 'react-test-renderer';
import Loadable from 'react-loadable';

beforeEach(async () => {
  await Loadable.preloadAll();
});

it('renders without crashing', async () => {
  const tree = TestRenderer.create(<App />);
  expect(tree.toJSON()).toMatchSnapshot();
});
