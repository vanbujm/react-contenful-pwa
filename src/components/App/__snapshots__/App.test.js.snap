// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
    </a>
    <a
      className="link"
      href="/content"
      onClick={[Function]}
    >
      Content
    </a>
  </header>
  <div
    className="layout"
  >
    <div>
      Home
    </div>
  </div>
</div>
`;
